enhancements/running-on-the-host.md

```diff
@@ -0,0 +1,167 @@
+- running the controller pod, in charge of reading the static configuration and receiving the additional configuration when the kubernete api is available
+
+#### Handling the lifecycle of the components
+
+The router and the contrller need to run as systemd units. This allows us to run them before the kubelet and the CNI start.
+Once the machinery is implemented, alternative ways to run them can be added for those environments where systemd is not enabled.
+
+#### Running on the host
+
+Podman provides a way to maintain the current pod structure while running the pods as systemd units. This allow us to keep reasoning of pods and
+to have a good parallel between the "Kubernetes mode" and the "Host mode".
+
+Because of this, the initial implementation will be based on Podman. As per systemd above, when we have a way of running the pods on the host, it should be
+relatively easy to adapt to other container runtimes.
+
+#### Interactions
+
+There are some interactions between the controller process and the router (and the surrounding system) that need to be adapted to the new scenario:
+
+##### Retrieving the router's network namespace
+
+We are using the container runtime api to retrieve the network namespace of the pod.
+When running on the host, the target namespace can be reitrieved by telling podman to fill a pid file, and by using the pid to retrieve the network namespace of the process.
+
+##### Sending a signal to the reloader container to reload the FRR configuration
+
+Currently the contrller retrieves the target pod's IP using the kubernetes API. When running on the host, a unix socket can be shared between the containers.
+
+##### Restarting the router pod
+
+The controller pod currently deletes the router pod to restart the underlay creation machinery. When running on the host, the controller pod can interact either with Podman or with Systemd to restart the pod.
+
+#### Consuming the static configuration
+
+The static configuration must be provided to the OpenPERouter as files in a known location. The configuration is composed by:
+
+- A static configuration with the same structure of the CRDs (ie underlays, vnis) to be used for bringing up the basic connectivity
+- An extra configuration to fill the requirements of running on the host. At the time of writing, the only extra configuration required seem to be a node index, to be associated manually (or by automation) to be different per each node.
+
+#### Consuming the Kubernetes API
+
+Running the controller on the host while it's still be able to access the kubernetes API is a challenge. One possible way to solve this is to have a proxy pod running on the cluster and translating its credentials to the controller running on the host.
+
+#### Merging the two configurations
+
+The static configuration must be treated exactly as configuration coming from the CRs. In this sense it must be validated, and the static configuration must be merged with the configuration read by CRs.
```

Comment:
What if the static configuration collides with a resource in the API server? 
It's likely an implementation detail, but the static configuration should be put into the API server, shouldn't it?

Reply:
my idea was to keep the static config unmutable. This is because that configuration is likely to be required for the basic connectivity, and any change is disruptive. Let me clarify this.
As a starter, I think it's more than valid. We can add what you are suggesting later if needed.